### DISCUSSION TITLE
[QUESTION] Appending chain to a chord executes the chain twice

### DISCUSSION THREAD

**Discussion by Janarcho** [celery/celery#7669] [Q&A]
# Checklist

* [ ] I have verified that the issue exists against the `master` branch of Celery.
* [ ] This has already been asked to the [discussions forum](https://github.com/celery/celery/discussions) first.
* [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
* [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
* [ ] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
* [ ] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
* [ ] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information
* [ ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
* [ ] I have verified that the issue exists against the `master` branch of Celery.
* [ ] I have included the contents of ``pip freeze`` in the issue.
* [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

* [x] I have tried reproducing the issue on more than one Python version
  and/or implementation.
* [x] I have tried reproducing the issue on more than one message broker and/or
  result backend.
* [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
* [x] I have tried reproducing the issue on more than one operating system.
* [ ] I have tried reproducing the issue on more than one workers pool.
* [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
* [x] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues
* None

#### Possible Duplicates
* None

## Environment & Settings

**Celery version**:

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
5.2.0 (dawn-chorus)
```

</p>
</details>

# Steps to Reproduce
## Required Dependencies

* **Minimal Python Version**: N/A or Unknown
* **Minimal Celery Version**: N/A or Unknown
* **Minimal Kombu Version**: N/A or Unknown
* **Minimal Broker Version**: N/A or Unknown
* **Minimal Result Backend Version**: N/A or Unknown
* **Minimal OS and/or Kernel Version**: N/A or Unknown
* **Minimal Broker Client Version**: N/A or Unknown
* **Minimal Result Backend Client Version**: N/A or Unknown

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
<p>

```

```

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

```python
# This executes "as expected" - first the group, then the tasks all one after the other ...
post_processing = (task_doing_nothing.si() | task_doing_nothing.si())
workflow = chain(
    group(
        task_doing_nothing.si(),
        task_doing_nothing.si(),
    ),
    task_doing_nothing.si()
) | post_processing
workflow.delay().get()

# This executes the post processing chain twice ... which I dont understand ...
post_processing = (task_doing_nothing.si() | task_doing_nothing.si())
workflow = chain(
    group(
        task_doing_nothing.si(),
        task_doing_nothing.si(),
    ),
    task_doing_nothing.si()
)
workflow |= post_processing
workflow.delay().get()
```

</p>
</details>

# Expected Behavior

I would expect no difference in the two approaches detailed above, i.e. I would not expect to post_processing chain to be executed twice.

# Actual Behavior

Using the second approach post_processing chain is executed twice.

[a comment by an automated account is not shown]

**Comment by Smixi** [comment]
I've runned your following code using a Redis broker/backend and it worked fine (not runned twice for the last task).
```python
@shared_task
def task_doing_nothing(*args, **kwargs):
    print("Done")
```
Output for both ways: 
```
                .> celery           exchange=celery(direct) key=celery
                

[2022-01-16 03:55:08,729: WARNING/ForkPoolWorker-16] Done
[2022-01-16 03:55:08,730: WARNING/ForkPoolWorker-1] Done
[2022-01-16 03:55:08,743: WARNING/ForkPoolWorker-16] Done
[2022-01-16 03:55:08,750: WARNING/ForkPoolWorker-1] Done
[2022-01-16 03:55:08,752: WARNING/ForkPoolWorker-16] Done
```
I'm on version 5.2.3 but I don't think there much added about this topic ?
